### ISSUE TITLE
REDMINE_DB_PORT not working

### ISSUE THREAD

**Issue by informatique4ipbs** [docker-library/redmine#130]
by modifying the docker-entrypoint.sh as follow, it works as written in the doc.

line 42 : file_env 'REDMINE_DB_PORT' "${MYSQL_ENV_MYSQL_USER:-3306}"
line 50 : file_env 'REDMINE_DB_PORT' "${POSTGRES_ENV_POSTGRES_PORT:-5432}"

by the way when i play the dockerfile for https://github.com/docker-library/redmine/blob/ba719a0b180982a45d994e0ab66ac7a4ea106bfe/3.4/Dockerfile it compains with bad gnupg keys for GOSU ...
hope this helps

**Comment by tianon** [comment]
Can you provide more details about what you're doing and what about it isn't working?

If you specify `REDMINE_DB_PORT` with an alternate port, that will be reflected in the generated `config/database.yml` file.

**Comment by informatique4ipbs** [comment]
Hi, 
for example if i launch ther redmine image with "-e REDMINE_DB_PORT=5434 -e REDMINE_DB_POSTGRES=10.59.64.18 " it continues to try to contact the machine 10.59.64.18:5432 and not 10.59.64.18:5434. With the modification i propose i works as expected (at least for me :-})

For the GOSU problem, juste take a fresh install of a debian stretch and a fresh install of docker.

**Comment by yosifkit** [comment]
1. I am unable to reproduce. Can you give the full commands of what exactly was run and any log output? Did you have any filled volumes before starting? (the parts in question only run if `config/database.yml` is not a file)

1. what was the exact error? The fingerprint for Tianon's key in the Dockerfile is the correct one. Most likely the error was `address not available` or `key XXXXXX can't be retrieved` (https://github.com/docker-library/httpd/issues/66 and https://github.com/docker-library/mysql/issues/263, respectively).
    - TLDR; GPG servers are extremely flaky (https://github.com/docker-library/official-images/issues/4252#issuecomment-381783035)
    - which we have solved on our servers by using a aggressive proxy (https://github.com/docker-library/php/pull/666)

**Comment by wglambert** [comment]
Since there isn't anything denoting an error with the image I'm going to close. If you do find something otherwise then I can re-open.

**Comment by MarcAntoineCRUE** [comment]
Hi,

This issue is quite old, however I face the same problem.
Using the variable REDMINE_DB_PORT does not have any effect.

I'm deploying a second redmine to makes some tests. Two redmine with two db (almost a simple copy of the db).

But i changed the port of the mysql on the second db, and tried to connect. but the two redmine instances are connecting to the first one (port 3306).

here is the content of my docker compose :

```
version: '3.8'
services:

  redmine:
    image: redmine:${REDMINE_VERSION}
    container_name: redmine
    restart: unless-stopped
    depends_on:
      - redmine_db
    environment:
      REDMINE_PLUGINS_MIGRATE: 1
      REDMINE_DB_MYSQL: redmine_db
      REDMINE_DB_DATABASE: ${MYSQL_DB_NAME}
      REDMINE_DB_USERNAME: ${MYSQL_USERNAME}
      REDMINE_DB_PASSWORD: ${MYSQL_PASSWORD}
      REDMINE_SECRET_KEY_BASE: ${REDMINE_TOKEN}
    volumes:
      - ./dataredmine/backup:/media/backup
      - ./dataredmine/redmine/files:/usr/src/redmine/files
      - ./dataredmine/redmine/log:/usr/src/redmine/log
      - ./dataredmine/redmine/plugins:/usr/src/redmine/plugins
      - ./dataredmine/redmine/public:/usr/src/redmine/public
      - ./dataredmine/redmine/config:/usr/src/redmine/config
    labels:
      # Network
      - traefik.enable=true
      # Https
      - traefik.http.routers.redmine.rule=Host(`redmine.${DOMAIN}`)
      - traefik.http.routers.redmine.entrypoints=websecure
      - traefik.http.routers.redmine.tls=true
      - traefik.http.routers.redmine.service=redmine
      # Loadbalancer
      - traefik.http.services.redmine.loadbalancer.server.port=3000
      # Middlewares
      - traefik.http.routers.redmine.middlewares=redmine-ssl
      - traefik.http.middlewares.redmine-ssl.headers.sslForceHost=true
      - traefik.http.middlewares.redmine-ssl.headers.sslHost=redmine.${DOMAIN}

  redmine_db:
    image: mysql:${MYSQL_VERSION}
    container_name: redmine_db
    restart: unless-stopped
    environment:
      MYSQL_DATABASE: ${MYSQL_DB_NAME}
      MYSQL_USER: ${MYSQL_USERNAME}
      MYSQL_PASSWORD: ${MYSQL_PASSWORD}
      MYSQL_ROOT_PASSWORD: ${MYSQL_ROOT_PASSWORD}
    volumes:
      - ./dataredmine/mysql:/var/lib/mysql
      - ./dataredmine/backup:/media/backup
    labels:
      - "traefik.enable=false"



  redmine_2:
    image: redmine:${REDMINE_VERSION}
    container_name: redmine_2
    restart: unless-stopped
    depends_on:
      - redmine_db_2
    ports:
     - 3001:3000
    environment:
      REDMINE_PLUGINS_MIGRATE: 1
      REDMINE_DB_MYSQL: redmine_db_2
      REDMINE_DB_DATABASE: ${MYSQL_DB_NAME}
      REDMINE_DB_USERNAME: ${MYSQL_USERNAME}
      REDMINE_DB_PASSWORD: ${MYSQL_PASSWORD}
      REDMINE_SECRET_KEY_BASE: ${REDMINE_TOKEN}
      REDMINE_DB_PORT: 3307
    volumes:
      - ./dataredmine2/backup:/media/backup
      - ./dataredmine2/redmine/files:/usr/src/redmine/files
      - ./dataredmine2/redmine/log:/usr/src/redmine/log
      - ./dataredmine2/redmine/plugins:/usr/src/redmine/plugins
      - ./dataredmine2/redmine/public:/usr/src/redmine/public
      - ./dataredmine2/redmine/config:/usr/src/redmine/config

  redmine_db_2:
    image: mysql:${MYSQL_VERSION}
    container_name: redmine_db_2
    restart: unless-stopped
    environment:
      MYSQL_DATABASE: ${MYSQL_DB_NAME}
      MYSQL_USER: ${MYSQL_USERNAME}
      MYSQL_PASSWORD: ${MYSQL_PASSWORD}
      MYSQL_ROOT_PASSWORD: ${MYSQL_ROOT_PASSWORD}
      MYSQL_TCP_PORT: 3307
    volumes:
      - ./dataredmine2/mysql:/var/lib/mysql
      - ./dataredmine2/backup:/media/backup
    labels:
      - "traefik.enable=false"

networks:
  default:
    external:
      name: Services

```

The redmine and redmine_2 containers are connecting to the redmine_db both side. When I'm setting a new issue on the first instance, the issue is also present on the second redmine instance, as they are using the same db.
